tensorboard/plugins/profile/profile_plugin.py

```diff
@@ -327,12 +329,32 @@ def data_route(self, request):
       return http_util.Respond(request, '404 Not Found', 'text/plain', code=404)
     return http_util.Respond(request, data, 'application/json')
 
+  @wrappers.Request.application
+  def capture_route(self, request):
+    service_addr = request.args.get('service_addr')
+    duration = int(request.args.get('duration'))
```

Comment:
if the arg `duration` is missing, this will raise TypeError. What is an expected outcome when certain inputs are missing?

Reply:
We don't allow user click "capture" if duration is missing.

Reply:
Should the server blow up in that case? Please handle the input properly.

Reply:
An exception will not blow up the server. It will just return an error response.